dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -254,6 +250,21 @@ ListenableFuture<Response> execute() {
             return result;
         }
 
+        private boolean requestCanBeRetried() {
+            return request.body()
+                    .map(body -> {
```

Comment:
I'd prefer to implement this without lambdas, avoiding a slight bit of overhead (the `map` function is a java object, which must be allocated and freed -- modern GCs are incredible, but not as efficient as avoiding allocations where it's relatively simple to do so).

In most places I'd prefer the way you've written this as is, but the constraints change a bit deeper in our infra.

Reply:
I see. Thanks.